discovery-infra/bootstrap_in_place.py

```diff
@@ -0,0 +1,171 @@
+    config["networking"]["machineNetwork"][0]["cidr"] = net_asset["machine_cidr"]
+
+    with open(INSTALL_CONFIG, "w") as _file:
+        yaml.dump(config, _file)
+
+
+def setup_files_and_folders(args, net_asset, cluster_name):
+    logging.info("Creating needed files and folders")
+    utils.recreate_folder(consts.BASE_IMAGE_FOLDER, force_recreate=False)
+    utils.recreate_folder(IBIP_DIR, with_chmod=False, force_recreate=True)
+    shutil.copy(os.path.join(RESOURCES_DIR, INSTALL_CONFIG_FILE_NAME), IBIP_DIR)
+    fill_install_config(args.pull_secret, args.ssh_key, net_asset, cluster_name)
+
+
+def str_presenter(dumper, data):
+    if "ssh-rsa" in data:  # check for multiline string
+        return dumper.represent_scalar('tag:yaml.org,2002:str', data, style='|')
+    return dumper.represent_scalar('tag:yaml.org,2002:str', data)
+
+
+def create_controller(net_asset):
+    return TerraformController(
+        cluster_name="test-infra-cluster",
+        num_masters=1,
+        num_workers=0,
+        master_memory=32 * 1024,  # 32GB of RAM
+        master_vcpu=12,
+        net_asset=net_asset,
+        iso_download_path="<TBD>",  # will be set later on
+        bootstrap_in_place=True,
+    )
+
+
+def all_operators_up():
+    statuses = get_operators_status(KUBE_CONFIG)
+    if not statuses:
+        logging.debug("No operator has been found currently...")
+        return False
+
+    invalid_operators = [operator for operator, up in statuses.items() if not up]
+
+    all_operators_are_valid = len(invalid_operators) == 0
+
+    if not all_operators_are_valid:
+        logging.debug("Following operators are still down: %s", ", ".join(invalid_operators))
+
+    return all_operators_are_valid
+
+
+def gather_sosreport_data(host_ip):
+    with SshConnection(ip=host_ip, private_ssh_key_path=SSH_KEY) as ssh:
+        ssh.upload_file(SOSREPORT_SCRIPT, "/tmp/man_sosreport.sh")
+        ssh.execute("chmod a+x /tmp/man_sosreport.sh")
+        ssh.execute("sudo /tmp/man_sosreport.sh")
+        ssh.download_file("/tmp/sosreport.tar.bz2", IBIP_DIR)
+
+
+def execute_ibip_flow(args):
+    host_ip = None
+    try:
+        openshift_release_image = os.getenv('OPENSHIFT_INSTALL_RELEASE_IMAGE')
+        if not openshift_release_image:
+            raise ValueError("os env OPENSHIFT_INSTALL_RELEASE_IMAGE must be provided")
+
+        net_asset = NetworkAssets().get()
+        controller = create_controller(net_asset)
+        setup_files_and_folders(args, net_asset, controller.cluster_name)
+
+        utils.extract_installer(openshift_release_image, BUILD_DIR)
+        installer_generate()
+
+        download_live_image(f"{BUILD_DIR}/installer-image.iso")
+        image_path = embed("installer-image.iso", "bootstrap.ign", EMBED_IMAGE_NAME)
+
+        logging.info("Starting nodes...")
+        controller.image_path = image_path
+        controller.start_all_nodes()
+        logging.info("Nodes started!")
+
+        logging.info("Configuring /etc/hosts...")
```

Comment:
can we have waiting for installation to complete function?

Reply:
yes